File: snippets/overview/snippet-02.ts
// Set before initializing the SDK
process.env.X_GLEAN_INCLUDE_EXPERIMENTAL = 'true';

import { Glean } from '@gleanwork/api-client';

const glean = new Glean({
  apiToken: process.env['GLEAN_API_TOKEN'] ?? '',
  serverURL: 'https://your-instance-be.glean.com',
});

File: snippets/index-documents/snippet-02.py
from glean.api_client import Glean
import os

with Glean(
    api_token=os.getenv("GLEAN_API_TOKEN", ""),
    server_url=os.getenv("GLEAN_SERVER_URL", ""),
) as client:
    # Index a document
    try:
        client.indexing.documents.index(request={
            "document": {
                "datasource": "gleantest",
                "objectType": "EngineeringDoc",
                "id": "blueskytest-1",
                "title": "This doc will help you get familiar with Blue Sky API",
                "viewURL": "https://bluesky.test/blueskytest-1",
                "body": {
                    "mimeType": "text/plain",
                    "textContent": "This doc will help you get familiar with Blue Sky API"
                },
                "permissions": {
                    "allowAnonymousAccess": True
                }
            }
        })
    except Exception as e:
        print(f"Exception when indexing document: {e}")
